src/peer_connection/mod.rs

```diff
@@ -765,8 +765,28 @@ impl RTCPeerConnection {
                     if !t.mid().await.is_empty() {
                         continue;
                     }
-                    let greater_mid = self.internal.greater_mid.fetch_add(1, Ordering::SeqCst);
-                    t.set_mid(format!("{}", greater_mid + 1)).await?;
+
+                    if let Some(gen) = &self.internal.setting_engine.mid_generator {
+                        let current_greatest = self.internal.greater_mid.load(Ordering::SeqCst);
+                        let mid = (gen)(current_greatest);
+
+                        // If it's possible to parse the returned mid as numeric, we will update the greater_mid field.
+                        let numeric_mid = match mid.parse::<isize>() {
+                            Ok(n) => n,
+                            Err(_) => continue,
+                        };
+
+                        t.set_mid(mid).await?;
+
+                        if numeric_mid > self.internal.greater_mid.load(Ordering::SeqCst) {
```

Comment:
This isn't done in the other case why do we need to do it here 🤔

Reply:
See above for incoming mids